src/site_apportionment/site_apportionment.py

```diff
@@ -1,98 +1,474 @@
-    dfc["site_weight"] = dfc["site_percent"] / dfc["site_percent_total"]
+    df = calculate_weights(df)
 
     # Remove unnecessary columns as they are no longer needed
-    dfc = dfc.drop(columns=["site_percent", "site_percent_total"], axis=1)
+    df = clean_up_post_calc(df)
+
+    return df
+
+
+# TODO: Move this to validation module and make it less specific to "NONE    "
+def clean_postcodes(df: pd.DataFrame, postcode_col: str) -> pd.DataFrame:
+    """
+    Cleans "NONE" postcodes by replacing them with an empty string.
+
+    Args:
+        df (pd.DataFrame): The input DataFrame.
+        postcode_col (str): The name of the postcode column.
+
+    Returns:
+        pd.DataFrame: The DataFrame with cleaned postcodes.
+    """
+    df.loc[df[postcode_col] == "NONE    ", postcode_col] = ""
+    return df
+
+
+def set_short_form_percentages(
+    df: pd.DataFrame, form_col: str, short_code: str, percent_col: str
+) -> pd.DataFrame:
+    """
+    Sets the percent column to 100 for short form records.
+    If the percent column for short forms is not blank, raises an error.
+
+    Args:
+        df (pd.DataFrame): The input DataFrame.
+        form_col (str): The name of the form column.
+        short_code (str): The code for short forms.
+        percent_col (str): The name of the percent column.
+
+    Returns:
+        pd.DataFrame: The DataFrame with updated percentages for short forms.
+
+    Raises:
+        ValueError: If the percent column for short forms is not blank.
+    """
+    short_forms = df[df[form_col] == short_code]
+    if not short_forms[percent_col].isna().all():
+        raise ValueError("Percent column for short forms should be blank.")
+
+    df.loc[df[form_col] == short_code, percent_col] = 100
+    return df
+
+
+def split_many_sites_df(
+    df: pd.DataFrame,
+    form_col: str,
+    long_code: str,
+    postcode_col: str,
+    instance_col: str,
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """
+    Splits the DataFrame into two based on certain conditions.
+
+    Args:
+        df (pd.DataFrame): The input DataFrame.
+        form_col (str): The name of the form column.
+        long_code (str): The code for long forms.
+        postcode_col (str): The name of the postcode column.
+        instance_col (str): The name of the instance column.
+
+    Returns:
+        Tuple[pd.DataFrame, pd.DataFrame]: A tuple containing two DataFrames.
+    """
+    # Condition of long forms, many sites, instance >=1, non-null postcodes
+    cond = (
+        (df[form_col] == long_code)
+        & (df[postcode_col + "_count"] > 1)
+        & (df[instance_col] >= 1)
+    )
+
+    # Dataframe many_sites_df with many products - for apportionment and Cartesian product
+    many_sites_df = df.copy()[cond]
+
+    # Dataframe with everything else - save unchanged
+    df_out = df[~cond]
+    df_out = df_out.drop(columns=[postcode_col + "_count"], axis=1)
+
+    return many_sites_df, df_out
+
+
+def spawn_column_lists(
```

Comment:
It's down to personal preference and opinion, but personally I wouldn't have created a func for these 2 lines of code... it takes 3 lines to call the function but the code itself would be 2 lines.

I can see it's called twice, I wonder whether it could be with the worth putting them in the "globals" section at the very top (if that's what you call it).